include/boost/geometry/iterators/flatten_iterator.hpp

```diff
@@ -215,7 +215,8 @@ class flatten_iterator
                 --m_outer_it;
             }
             while ( empty(m_outer_it) );
-            m_inner_it = --AccessInnerEnd::apply(*m_outer_it);
+            m_inner_it = AccessInnerEnd::apply(*m_outer_it);
+            --m_inner_it;
```

Comment:
For this one I think we should leave it as it is and modify how this is called. The `AccessInnerEnd` class should return a const iterator instead of a mutable iterator, and this should take care of it.